File: languages/php/Bloock/Entity/Identity/DecimalAttribute.php
<?php

namespace Bloock\Entity\Identity;

/**
 * Represents an attribute with a decimal value.
 */
class DecimalAttribute extends Attribute
{
    public static function fromProto(\Bloock\DecimalAttribute $res): DecimalAttribute
    {
        return new DecimalAttribute($res->getId(), $res->getValue());
    }

    public function toProto(): \Bloock\DecimalAttribute
    {
        $p = new \Bloock\DecimalAttribute();
        $p->setId($this->id);
        $p->setValue($this->value);
        return $p;
    }
}

File: languages/python/bloock/entity/identity/decimal_attribute.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_pb2 as proto
from bloock.entity.identity.attribute import Attribute


class DecimalAttribute(Attribute):
    """
    Represents an attribute with a decimal value.
    """
    @staticmethod
    def from_proto(a: proto.DecimalAttribute) -> DecimalAttribute:
        return DecimalAttribute(
            id=a.id,
            value=a.value,
        )

    def to_proto(self) -> proto.DecimalAttribute:
        return proto.DecimalAttribute(
            id=self.id,
            value=self.value,
        )